From be7739e64174bdce963a81da6350ff4f0b56b731 Mon Sep 17 00:00:00 2001
From: Adam Duskett <Aduskett@rivian.com>
Date: Wed, 24 Feb 2021 20:03:21 -0800
Subject: [PATCH] add dracut to initramfs infrastructure

As dracut also creates a cpio file, the initramfs and linux.mk files can
now support dracut as well.

Signed-off-by: Adam Duskett <Aduskett@rivian.com>
---
 fs/initramfs/Config.in    |  2 +-
 fs/initramfs/initramfs.mk | 12 +++++++++++-
 linux/linux.mk            |  2 +-
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/fs/initramfs/Config.in b/fs/initramfs/Config.in
index 9d5a3f92e6..3ead5a1ceb 100644
--- a/fs/initramfs/Config.in
+++ b/fs/initramfs/Config.in
@@ -1,7 +1,7 @@
 config BR2_TARGET_ROOTFS_INITRAMFS
 	bool "initial RAM filesystem linked into linux kernel"
 	depends on BR2_LINUX_KERNEL
-	select BR2_TARGET_ROOTFS_CPIO
+	select BR2_TARGET_ROOTFS_CPIO if !BR2_TARGET_ROOTFS_DRACUT
 	help
 	  Integrate the root filesystem generated by Buildroot as an
 	  initramfs inside the kernel image. This integration will
diff --git a/fs/initramfs/initramfs.mk b/fs/initramfs/initramfs.mk
index 3b3d4ed8b9..c5c1e4e267 100644
--- a/fs/initramfs/initramfs.mk
+++ b/fs/initramfs/initramfs.mk
@@ -33,4 +33,14 @@ endif
 # Not using the rootfs infra, so fake the variables
 ROOTFS_INITRAMFS_NAME = rootfs-initramfs
 ROOTFS_INITRAMFS_TYPE = rootfs
-ROOTFS_INITRAMFS_DEPENDENCIES = rootfs-cpio linux
+ROOTFS_INITRAMFS_DEPENDENCIES = linux
+
+ifeq ($(BR2_TARGET_ROOTFS_CPIO),y)
+ROOTFS_INITRAMFS_DEPENDENCIES += rootfs-cpio
+endif
+
+ifeq ($(BR2_TARGET_ROOTFS_DRACUT),y)
+ROOTFS_INITRAMFS_DEPENDENCIES += rootfs-dracut
+endif
+
+
diff --git a/linux/linux.mk b/linux/linux.mk
index 5e4b319cf1..45b6286926 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -579,7 +579,7 @@ $(eval $(kconfig-package))
 .PHONY: linux-rebuild-with-initramfs
 linux-rebuild-with-initramfs: $(LINUX_DIR)/.stamp_target_installed
 linux-rebuild-with-initramfs: $(LINUX_DIR)/.stamp_images_installed
-linux-rebuild-with-initramfs: rootfs-cpio
+linux-rebuild-with-initramfs: rootfs-cpio rootfs-dracut
 linux-rebuild-with-initramfs:
 	@$(call MESSAGE,"Rebuilding kernel with initramfs")
 	# Build the kernel.
-- 
2.25.1

